onnx/defs/tensor/defs.cc

```diff
@@ -806,68 +810,106 @@ Example 2:
           [4.5, 5.9],
       ],
   ]
+Example 3:
+  data = [
+      [1, 2, 3],
+      [4, 5, 6],
+      [7, 8, 9],
+  ]
+  indices = [
+      [1, 2, 0],
+      [2, 0, 0],
+  ]
+  output = [
+      [
+          [4, 8, 3],
+          [7, 2, 3],
+      ],
+  ]
 )DOC";
 
 ONNX_OPERATOR_SET_SCHEMA(
     Gather,
-    1,
+    10,
     OpSchema()
-        .SetDoc(Gather_ver1_doc)
+        .SetDoc(Gather_ver10_doc)
         .Attr(
             "axis",
             "Which axis to gather on. Negative value means "
-            "counting dimensions from the back. Accepted range in [-r, r-1]",
+            "counting dimensions from the back. Accepted range in [-r, r-1]. ",
+            AttributeProto::INT,
+            static_cast<int64_t>(0))
+        .Attr(
+            "elem_index",
```

Comment:
Maybe we can consider a more evocative name for this attribute. Something like, `GatherElement` or `GatherSlice`, or something else.

Reply:
I am not sure I understand the description of the attribute below. It sounds like the main distinction is whether the same indices tensor is used for all elements of the non-indexed axis (the original mode) or whether a different indices (sub)tensor is used for each element of the non-indexed axis (the new mode). Is this correct? If so, may be an attribute like "SharedIndices" or "SameIndices" might help.